### PR TITLE
[specific ci=Group1-Docker-Commands] Add support for network labels

### LINKED ISSUE #3738: docker network --label is not supported
**VIC version:**

0.8

**Deployment details:**

What was the vic-machine create command used to deploy the VCH?
bin/vic-machine-linux create --name=vch-3 --target=root:pwd@192.168.60.162 --thumbprint=... --volume-store=datastore1/test:default --no-tlsverify --no-tls

**Steps to reproduce:**
docker -H 192.168.60.130:2375 network create --label="aaa=bbb" test1

**Actual behavior:**
```
docker -H 192.168.60.130:2375 network inspect test1
[
    {
        "Name": "test1",
        "Id": "f02bfaf65276440d951f17d5a7a80747b0d8168ff3dfad7928901d2b04fd9025",
        "Scope": "",
        "Driver": "bridge",
        "EnableIPv6": false,
        "IPAM": {
            "Driver": "",
            "Options": {},
            "Config": [
                {
                    "Subnet": "172.19.0.0/16",
                    "Gateway": "172.19.0.1"
                }
            ]
        },
        "Internal": false,
        "Containers": {},
        "Options": {},
        "Labels": {}
    }
]
```
**Expected behavior:**
```
"Labels": {
            "aaa": "bbb"
        }
```

This is needed by the "label" syntax in the docker-compose file. See https://github.com/vmware/vic/issues/3635#issuecomment-275496609

### PR DESCRIPTION
**Pull request by anchal-agrawal**
Adds support for labels in `network create` and `network inspect`. With this change, `network ls -f label=...` also works as expected.

Fixes #3738

### CODE REVIEW

**Comment by anchal-agrawal** [comment]
@hmahmood @sflxn The code changes for label support are in these files:
```
lib/apiservers/engine/backends/network.go
lib/apiservers/portlayer/restapi/handlers/scopes_handlers.go
lib/portlayer/network/scope.go
```

The other changes are just a refactor to improve readability by replacing 5+ function input args with one struct containing those fields.

File: lib/portlayer/network/context.go [hunk 1]
```diff
@@ -133,7 +137,17 @@ func NewContext(config *Configuration, kv kvstore.KeyValueStore) (*Context, erro
 		}
 
 		subnet := net.IPNet{IP: n.Gateway.IP.Mask(n.Gateway.Mask), Mask: n.Gateway.Mask}
-		s, err := ctx.newScope(n.Type, nn, &subnet, n.Gateway.IP, n.Nameservers, pools)
+
+		scopeData = ScopeData{
```
Comment (hmahmood) [inline lib/portlayer/network/context.go]: This should be a pointer.

File: lib/portlayer/network/context.go [hunk 2]
```diff
@@ -310,44 +324,48 @@ func (c *Context) addScope(s *Scope) error {
 	return nil
 }
 
-func (c *Context) newScopeCommon(id uid.UID, name, scopeType string, subnet *net.IPNet, gateway net.IP, dns []net.IP, pools []string, network object.NetworkReference) (*Scope, error) {
+func (c *Context) newScopeCommon(id uid.UID, scopeType string, network object.NetworkReference, scopeData ScopeData) (*Scope, error) {
```
Comment (hmahmood) [inline lib/portlayer/network/context.go]: Make scopeData a pointer.

File: lib/portlayer/network/context.go (hunk 1 above)
Comment (anchal-agrawal) [inline lib/portlayer/network/context.go]: Fixed all occurrences. Thanks!

**Pull request merged**